snap/snapenv/snapenv_test.go

```diff
@@ -108,14 +110,31 @@ func (ts *HTestSuite) TestUser(c *C) {
 }
 
 func (ts *HTestSuite) TestUserForClassicConfinement(c *C) {
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("/")
+	c.Assert(os.MkdirAll(dirs.FeaturesDir, 0755), IsNil)
+
+	// With the classic-preserves-xdg-runtime-dir feature disabled the snap
+	// per-user environment contains an override for XDG_RUNTIME_DIR.
 	env := userEnv(mockClassicSnapInfo, "/root")
+	c.Assert(env, DeepEquals, map[string]string{
+		// NOTE: Both HOME and XDG_RUMTIME_DIR are not defined here.
+		"SNAP_USER_COMMON": "/root/snap/foo/common",
+		"SNAP_USER_DATA":   "/root/snap/foo/17",
+		"XDG_RUNTIME_DIR":  fmt.Sprintf(dirs.GlobalRootDir+"/run/user/%d/snap.foo", sys.Geteuid()),
+	})
 
+	// With the classic-preserves-xdg-runtime-dir feature enabled the snap
+	// per-user environment contains no overrides for XDG_RUNTIME_DIR.
+	f := features.ClassicPreservesXdgRuntimeDir
+	c.Assert(ioutil.WriteFile(f.ControlFile(), nil, 0644), IsNil)
+	env = userEnv(mockClassicSnapInfo, "/root")
 	c.Assert(env, DeepEquals, map[string]string{
-		// NOTE HOME Is absent! we no longer override it
+		// NOTE: Both HOME and XDG_RUMTIME_DIR are not defined here.
```

Comment:
```suggestion
		// NOTE: Both HOME and XDG_RUNTIME_DIR are not defined here.
```
I don't think it's rumtime here (yet)

Reply:
Lol :D